packages/@aws-cdk/aws-codepipeline/lib/jenkins-provider.ts

```diff
@@ -0,0 +1,256 @@
+    return new ImportedJenkinsProvider(parent, id, props);
+  }
+
+  public readonly abstract providerName: string;
+  public readonly abstract serverUrl: string;
+  public readonly abstract category: JenkinsProviderCategory;
+  public readonly version: string;
+
+  protected constructor(parent: cdk.Construct, id: string, version?: string) {
+    super(parent, id);
+
+    this.version = version || '1';
+  }
+
+  public export(): ImportJenkinsProviderProps {
+    return {
+      providerName: new cdk.Output(this, 'JenkinsProviderName', {
+        value: this.providerName,
+      }).makeImportValue().toString(),
+      serverUrl: new cdk.Output(this, 'JenkinsServerUrl', {
+        value: this.serverUrl,
+      }).makeImportValue().toString(),
+      version: new cdk.Output(this, 'JenkinsProviderVersion', {
+        value: this.version,
+      }).makeImportValue().toString(),
+      category: this.category,
+    };
+  }
+}
+
+class ImportedJenkinsProvider extends JenkinsProvider {
+  public readonly providerName: string;
+  public readonly serverUrl: string;
+  public readonly category: JenkinsProviderCategory;
+
+  constructor(parent: cdk.Construct, id: string, props: ImportJenkinsProviderProps) {
+    super(parent, id, props.version);
+
+    this.providerName = props.providerName;
+    this.serverUrl = props.serverUrl;
+    this.category = props.category;
+  }
+}
+
+export abstract class BaseJenkinsProvider extends JenkinsProvider {
+  public readonly providerName: string;
+  public readonly serverUrl: string;
+  public readonly category: JenkinsProviderCategory;
+
+  protected constructor(parent: cdk.Construct, id: string, category: JenkinsProviderCategory, props: JenkinsProviderProps) {
+    super(parent, id, props.version);
+
+    this.providerName = props.providerName;
+    this.serverUrl = props.serverUrl;
+    this.category = category;
+
+    new CustomActionRegistration(this, 'RegistrationResource', {
+      category: jenkinsCategory2ActionCategory(this.category),
+      artifactBounds: jenkinsArtifactsBounds,
+      provider: this.providerName,
+      version: this.version,
+      entityUrl: appendToUrl(props.serverUrl, 'job/{Config:ProjectName}'),
+      executionUrl: appendToUrl(props.serverUrl, 'job/{Config:ProjectName}/{ExternalExecutionId}'),
+      actionProperties: [
+        {
+          name: 'ProjectName',
+          required: true,
+          key: true,
+          queryable: true,
+        },
+      ],
+    });
+  }
+}
+
+/**
+ * Creation properties of Jenkins providers.
+ *
+ * @see JenkinsBuildProvider
+ * @see JenkinsTestProvider
+ */
+export interface JenkinsProviderProps {
+  /**
+   * The name of the Jenkins provider that you set in the AWS CodePipeline plugin configuration of your Jenkins project.
+   *
+   * @example 'MyJenkinsProvider'
+   */
+  providerName: string;
+
+  /**
+   * The base URL of your Jenkins server.
+   *
+   * @example 'http://myjenkins.com:8080'
+   */
+  serverUrl: string;
```

Comment:
Where is the authentication?

Reply:
There is no auth (CodePipeline does not talk to the Jenkins server at all) - this is strictly for displaying links in the UI.